MicroSplit noise model config changes for Lightning API

CatEek opened the pull request:
## Description

<!-- This section provides the necessary background and information for reviewers to
understand the code and have the correct mindset when examining changes. -->

> [!NOTE]  
> **tldr**: <!-- Write a one sentence summary. -->

MicroSplit config for Lightning API draft

### Background - why do we need this PR?

<!-- What problem are you solving? Describe in a few sentences the state before
this PR. Use code examples if useful. -->

Because we want MicroSplit to be there

### Overview - what changed?

<!-- What aspects and mechanisms of the code base changed? Describe only the general 
idea and overarching features. -->

Added noise model and likelihoods, relevant changes in training/validation step in lightning module.



**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [ ] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

CatEek committed:
init

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
hdn min conf

CatEek committed:
merge

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
format

CatEek committed:
tests pass

CatEek committed:
Merge remote-tracking branch 'origin/iz/refac/lightning_api_create_model' into iz/refac/lightning_api_create_model

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
mypy fix

CatEek committed:
Merge remote-tracking branch 'origin/iz/refac/lightning_api_create_model' into iz/refac/lightning_api_create_model

CatEek committed:
pred func fix

CatEek committed:
microsplit config + tests

CatEek committed:
microsplit config + tests

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
tests

CatEek committed:
tests

CatEek committed:
Merge remote-tracking branch 'origin/iz/feat/microsplit_lightning_api' into iz/feat/microsplit_lightning_api

CatEek committed:
config draft w/o data stats

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
data_stats fix

CatEek committed:
Merge remote-tracking branch 'origin/iz/feat/nm_ll_refac' into iz/feat/nm_ll_refac

jdeschamps reviewed (changes requested):
This PR is hard to review (and review was not requested?), I just went commit by commit since https://github.com/CAREamics/careamics/pull/512. It does not seem to have much and I don't fully understand what is happening, so not much to say. ^^

I do have two comments though:

- Can you change the PR description and name to reflect what the PR is about? My understanding is that it is simply changing the data stats defaults and calls in MicroSplit. So nothing to do with the config for Lightning API.
- There is an error raised during tests:

```python
=========================== short test summary info ============================
FAILED tests/config/test_configuration_factories.py::test_microsplit_configuration - TypeError: create_microsplit_configuration() got an unexpected keyword argument 'data_stats'
= 1 failed, 940 passed, 123 skipped, 506 deselected, 685 warnings in 89.95s (0:01:29) =
Error: Process completed with exit code 1. 
```

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.16.0,>=0.12.3 to >=0.12.3,<=0.17.3 (#529)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h3>Docs</h3>
<ul>
<li>📝 Remove duplicated line in <code>environment-variables.md</code>.
PR <a
href="https://redirect.github.com/fastapi/typer/pull/1277">#1277</a> by
<a href="https://github.com/neirzhei"><code>@​neirzhei</code></a>.</li>
<li>📝 Fix reference to <code>count</code> parameter in the
documentation. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1201">#1201</a> by
<a
href="https://github.com/PokkaKiyo"><code>@​PokkaKiyo</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.11.13 to 0.12.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1276">#1276</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1247">#1247</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Bump actions/checkout from 4 to 5. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1271">#1271</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump mkdocs-macros-plugin from 1.3.7 to 1.3.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1272">#1272</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump tiangolo/latest-changes from 0.3.2 to 0.4.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1265">#1265</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump pillow from 11.2.1 to 11.3.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1249">#1249</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1206">#1206</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/a5e41c53783558591923337d6a9579f7f64e1152"><code>a5e41c5</code></a>
🔖 Release version 0.17.3</li>
<li><a
href="https://github.com/fastapi/typer/commit/45cae190db45817343b00ff8b403b135f34c658f"><code>45cae19</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/0b8fa09bb6b5ed5f9ed5e26c0e8a7c4eb984226d"><code>0b8fa09</code></a>
✨ Allow annotated parsing with a subclass of <code>Path</code> (<a
href="https://redirect.github.com/fastapi/typer/issues/1183">#1183</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f0334c2a1081b3ffcb99bc4c42d85cc9d360c0e0"><code>f0334c2</code></a>
🔖 Release version 0.17.2</li>
<li><a
href="https://github.com/fastapi/typer/commit/73aca3bf87dfb5331486a603a01a1980299f38a5"><code>73aca3b</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/cdc29e27eff577403fc46b4f2d3dc912d975ad66"><code>cdc29e2</code></a>
🐛 Avoid printing <code>default: None</code> in the help section when
using Rich (<a
href="https://redirect.github.com/fastapi/typer/issues/1120">#1120</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/50b1018a81cb3f99534683e71b366df90cfe5d13"><code>50b1018</code></a>
🔖 Release version 0.17.1</li>
<li><a
href="https://github.com/fastapi/typer/commit/4e811192ac56edd1921de8b846ddf6145b9fe27e"><code>4e81119</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4d1cfd2698ec8e4809e58b51b1b32398970aa512"><code>4d1cfd2</code></a>
🐛 Fix markdown formatting in <code>--help</code> output (<a
href="https://redirect.github.com/fastapi/typer/issues/815">#815</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/83eaba8d6822023b4d46212c8ef75bbd707cc49d"><code>83eaba8</code></a>
🔖 Release version 0.17.0</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.3">compare
view</a></li>
</ul>
</details>
<br />

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.16.0,>=0.12.3 to >=0.12.3,<=0.17.3 (#529)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h3>Docs</h3>
<ul>
<li>📝 Remove duplicated line in <code>environment-variables.md</code>.
PR <a
href="https://redirect.github.com/fastapi/typer/pull/1277">#1277</a> by
<a href="https://github.com/neirzhei"><code>@​neirzhei</code></a>.</li>
<li>📝 Fix reference to <code>count</code> parameter in the
documentation. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1201">#1201</a> by
<a
href="https://github.com/PokkaKiyo"><code>@​PokkaKiyo</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.11.13 to 0.12.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1276">#1276</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1247">#1247</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Bump actions/checkout from 4 to 5. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1271">#1271</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump mkdocs-macros-plugin from 1.3.7 to 1.3.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1272">#1272</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump tiangolo/latest-changes from 0.3.2 to 0.4.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1265">#1265</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump pillow from 11.2.1 to 11.3.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1249">#1249</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1206">#1206</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/a5e41c53783558591923337d6a9579f7f64e1152"><code>a5e41c5</code></a>
🔖 Release version 0.17.3</li>
<li><a
href="https://github.com/fastapi/typer/commit/45cae190db45817343b00ff8b403b135f34c658f"><code>45cae19</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/0b8fa09bb6b5ed5f9ed5e26c0e8a7c4eb984226d"><code>0b8fa09</code></a>
✨ Allow annotated parsing with a subclass of <code>Path</code> (<a
href="https://redirect.github.com/fastapi/typer/issues/1183">#1183</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f0334c2a1081b3ffcb99bc4c42d85cc9d360c0e0"><code>f0334c2</code></a>
🔖 Release version 0.17.2</li>
<li><a
href="https://github.com/fastapi/typer/commit/73aca3bf87dfb5331486a603a01a1980299f38a5"><code>73aca3b</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/cdc29e27eff577403fc46b4f2d3dc912d975ad66"><code>cdc29e2</code></a>
🐛 Avoid printing <code>default: None</code> in the help section when
using Rich (<a
href="https://redirect.github.com/fastapi/typer/issues/1120">#1120</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/50b1018a81cb3f99534683e71b366df90cfe5d13"><code>50b1018</code></a>
🔖 Release version 0.17.1</li>
<li><a
href="https://github.com/fastapi/typer/commit/4e811192ac56edd1921de8b846ddf6145b9fe27e"><code>4e81119</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4d1cfd2698ec8e4809e58b51b1b32398970aa512"><code>4d1cfd2</code></a>
🐛 Fix markdown formatting in <code>--help</code> output (<a
href="https://redirect.github.com/fastapi/typer/issues/815">#815</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/83eaba8d6822023b4d46212c8ef75bbd707cc49d"><code>83eaba8</code></a>
🔖 Release version 0.17.0</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.3">compare
view</a></li>
</ul>
</details>
<br />

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.17.3,>=0.12.3 to >=0.12.3,<=0.17.4 (#531)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/1b50007284ccb38e71a5752b09b2cd3ca053d98e"><code>1b50007</code></a>
🔖 Release version 0.17.4</li>
<li><a
href="https://github.com/fastapi/typer/commit/90aab51837bbf0cf11c314c0fcbdd1e55f84d29a"><code>90aab51</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/263e42769f0d54b986275a800227c717c18660ea"><code>263e427</code></a>
🐛 Make sure <code>rich.markup</code> is imported when rendering help
text (<a
href="https://redirect.github.com/fastapi/typer/issues/1290">#1290</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f99defb7cad881190cae7c60c617bf4e1619bc43"><code>f99defb</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/f116fca2485a3d59cd879a07a74b8b92c028b4a3"><code>f116fca</code></a>
⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0 (<a
href="https://redirect.github.com/fastapi/typer/issues/1292">#1292</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/9c5c5437dde460c2cf712df4198ea9eeaab64c28"><code>9c5c543</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4c2129325a1014241a663f99f482a12af6868afc"><code>4c21293</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/bdf471038d6198bb67638b17e8d03fe0a07045d8"><code>bdf4710</code></a>
⬆ Bump ruff from 0.12.10 to 0.12.11 (<a
href="https://redirect.github.com/fastapi/typer/issues/1283">#1283</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/e8e472d1048c097cc69f5152d8146c4ba4d67a00"><code>e8e472d</code></a>
⬆ [pre-commit.ci] pre-commit autoupdate (<a
href="https://redirect.github.com/fastapi/typer/issues/1288">#1288</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/075eb3020706c136c56a1899d8d02d50d022ec9a"><code>075eb30</code></a>
📝 Update release notes</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.4">compare
view</a></li>
</ul>
</details>
<br />


Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.17.3,>=0.12.3 to >=0.12.3,<=0.17.4 (#531)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/1b50007284ccb38e71a5752b09b2cd3ca053d98e"><code>1b50007</code></a>
🔖 Release version 0.17.4</li>
<li><a
href="https://github.com/fastapi/typer/commit/90aab51837bbf0cf11c314c0fcbdd1e55f84d29a"><code>90aab51</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/263e42769f0d54b986275a800227c717c18660ea"><code>263e427</code></a>
🐛 Make sure <code>rich.markup</code> is imported when rendering help
text (<a
href="https://redirect.github.com/fastapi/typer/issues/1290">#1290</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f99defb7cad881190cae7c60c617bf4e1619bc43"><code>f99defb</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/f116fca2485a3d59cd879a07a74b8b92c028b4a3"><code>f116fca</code></a>
⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0 (<a
href="https://redirect.github.com/fastapi/typer/issues/1292">#1292</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/9c5c5437dde460c2cf712df4198ea9eeaab64c28"><code>9c5c543</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4c2129325a1014241a663f99f482a12af6868afc"><code>4c21293</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/bdf471038d6198bb67638b17e8d03fe0a07045d8"><code>bdf4710</code></a>
⬆ Bump ruff from 0.12.10 to 0.12.11 (<a
href="https://redirect.github.com/fastapi/typer/issues/1283">#1283</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/e8e472d1048c097cc69f5152d8146c4ba4d67a00"><code>e8e472d</code></a>
⬆ [pre-commit.ci] pre-commit autoupdate (<a
href="https://redirect.github.com/fastapi/typer/issues/1288">#1288</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/075eb3020706c136c56a1899d8d02d50d022ec9a"><code>075eb30</code></a>
📝 Update release notes</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.4">compare
view</a></li>
</ul>
</details>
<br />


Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

CatEek committed:
review comms + tests fix

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
rm params + todos

CatEek committed:
Merge remote-tracking branch 'origin/iz/feat/microsplit_lightning_api' into iz/feat/microsplit_lightning_api

CatEek committed:
Merge remote-tracking branch 'origin/main' into iz/feat/nm_ll_refac

CatEek committed:
Merge remote-tracking branch 'origin/iz/feat/microsplit_lightning_api' into iz/feat/nm_ll_refac

CatEek committed:
tests + precommits

CatEek committed:
Merge remote-tracking branch 'origin/main' into iz/feat/nm_ll_refac

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

jdeschamps commented on src/careamics/config/__init__.py:
```diff
@@ -13,6 +13,7 @@
     "DataConfig",
     "GaussianMixtureNMConfig",
     "HDNAlgorithm",
+    "HDNAlgorithm",
```
Duplicated line
```suggestion
```

jdeschamps commented on src/careamics/lightning/predict_data_module.py:
```diff
@@ -227,7 +227,7 @@ def predict_dataloader(self) -> DataLoader:
 def create_predict_datamodule(
     pred_data: Union[str, Path, NDArray],
-    data_type: Union[Literal["array", "tiff", "custom"], SupportedData],
+    data_type: Union[Literal["array", "tiff", "czi", "custom"], SupportedData],
```
CZI should be removed
```suggestion
    data_type: Union[Literal["array", "tiff", "custom"], SupportedData],
```

jdeschamps reviewed (approved):
In principle, I approve but do implement the very small requests (they can be committed directly through gh interface).

jdeschamps commented:
There are a few line too long ruff errors.

CatEek committed:
Update src/careamics/config/__init__.py

Co-authored-by: Joran Deschamps <6367888+jdeschamps@users.noreply.github.com>

CatEek committed:
Update src/careamics/lightning/predict_data_module.py

Co-authored-by: Joran Deschamps <6367888+jdeschamps@users.noreply.github.com>

CatEek committed:
Merge remote-tracking branch 'origin/main' into iz/feat/nm_ll_refac
